plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java

```diff
@@ -244,14 +247,24 @@ public void put(final InputStream inputStream) throws IOException, NugetPackageE
       recordMetadata.put(CREATED, creationTime);
       recordMetadata.put(LAST_UPDATED, creationTime);
       recordMetadata.put(PUBLISHED, creationTime);
-
+      OrientVertex component = null;
       try (InputStream in = tempStream.get()) {
-        createOrUpdatePackage(storageTx, recordMetadata, in);
+        component = createOrUpdatePackage(storageTx, recordMetadata, in);
       }
 
-      maintainAggregateInfo(storageTx, recordMetadata.get(ID));
+      String id = recordMetadata.get(ID);
+      String version = recordMetadata.get(VERSION);
+      maintainAggregateInfo(storageTx, id);
 
+      boolean isNew = component.getIdentity().isNew();  // must check before commit
       storageTx.commit();
+
+      if (isNew) {
+        getEventBus().post(new ComponentCreatedEvent(component, getRepository()));
+      }
+      else {
+        getEventBus().post(new ComponentUpdatedEvent(component, getRepository()));
+      }
     }
```

Comment:
No idea actually, but the assumption is that everything has successfully persisted once the transaction has been committed, and if the component is 'detached' and required additional effort for reading that would be the responsibility of the receiver.